Use web address when appropriate for a jump host

rosstimothy opened the pull request:
Determines whether the jump host provided via `tsh ssh -J` is belongs to the Proxy SSH or Web server to ensure when using jump hosts that connections are established directly on the target cluster.

Closes #25178

rosstimothy committed:
Use web address when appropriate for a jump hosts

Determines whether the jump host provided via `tsh ssh -J` is belongs
to the Proxy SSH or Web server to ensure when using jump hosts that
connections are established directly on the target cluster.

Closes #25178

Joerger reviewed (commented):
This fixed the issue for me. It would be good if we can add a regression test / fix `TestSSH/ssh_jump_host_access` since I expected this would be caught there.

rosstimothy committed:
Modify tsh tests to capture issues with jump hosts

Alters the root and leaf cluster and node names used by tsh tests
so that the root cluster is named `root` instead of `localhost` and
sets a unique `NodeName` for each cluster instead of reusing
`localnode` for both. This was masking problems in jump hosts tests
by connecting to the node in the root cluster instead of the leaf
cluster.

Some additional changes to tsh tests were made as a result of
changing the cluster and node names.

rosstimothy committed:
fix proxy client tests

rosstimothy committed:
update TestList to login once

rosstimothy committed:
ignore TestList in flaky test detector
